core/src/main/java/com/microsoft/applicationinsights/internal/config/JaxbAppInsightsConfigurationBuilder.java

```diff
@@ -54,25 +58,10 @@ public ApplicationInsightsXmlConfiguration build(InputStream resourceFile) {
                 return null;
             }
 
-            ApplicationInsightsXmlConfiguration applicationInsights = (ApplicationInsightsXmlConfiguration)unmarshaller.unmarshal(resourceFileReader);
-
-            return applicationInsights;
+            return (ApplicationInsightsXmlConfiguration)unmarshaller.unmarshal(resourceFileReader);
         } catch (JAXBException e) {
-            if (e.getCause() != null) {
-                InternalLogger.INSTANCE.error("Failed to parse configuration file: '%s'",
-                        ExceptionUtils.getStackTrace(e));
-            } else {
-                InternalLogger.INSTANCE.error("Failed to parse configuration file: '%s'",
-                        ExceptionUtils.getStackTrace(e));
-            }
-        } catch (NullPointerException e) {
-            e.printStackTrace();
-        } finally {
-            try {
-                resourceFile.close();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
+            InternalLogger.INSTANCE.error("Failed to parse configuration file: '%s'", ExceptionUtils.getStackTrace(e));
```

Comment:
I think this is before InternalLogger is initialized, so won't log?

Reply:
yes, but there you can turn on the logger with system properties instead of config. There are a few places where this occurs.

Reply:
should we enable logger via system properties during smoke tests?

Reply:
yeah, good call.